LanguageFeatures/Extension-types/exhaustiveness_enum_A01_t04.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with enum as a matched type are
+/// always exhaustive
+///
+/// @description Check that it is no compile-time error if a matched type of a
+/// switch expression is an extension type with an enum as a representation type
+/// and the set of cases is exhaustive
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+enum E {
+  a, b, c
+}
+
+extension type ET1(E _) {}
+extension type ET2(E _) implements E {}
+
+String testStatement1(ET1 e) {
+  switch (e) {
+    case E.a:
+    case E.a:
```

Comment:
Presumably this would cause an "unreachable case" warning or something like that. I guess that doesn't cause any trouble with any test run? What's the purpose of having two copies of `E.a`?

Reply:
It was made by purpose. What if this warning will affect exhaustiveness somehow? Let's be sure that not

Reply:
We could also simply remove the duplicate case: We're testing that a matched value type for the purpose of exhaustiveness is treated as its erasure, we aren't testing that switch statements have correct analysis of 'unreachable case' for a bunch of enum values (because we would have that test already, somewhere else).

Reply:
Removed. Please review